src/core/server/http/http_server.ts

```diff
@@ -251,7 +253,10 @@ export class HttpServer {
     );
 
     this.server.auth.scheme('login', () => ({
-      authenticate: adoptToHapiAuthFormat(fn, this.authState.set),
+      authenticate: adoptToHapiAuthFormat(fn, (req, { state, headers }) => {
+        this.authState.set(req, state);
+        this.authHeaders.set(req, headers);
```

Comment:
I used 2 different storages because they have different use cases

- the state may be exposed to any downstream plugin
- headers are meant to be used only by core when call elasticsearch cluster